counsel.el

```diff
@@ -731,6 +731,25 @@ With a prefix arg, restrict list to variables defined using
       (when doc
         (lv-delete-window)))))
 
+;;** `counsel-tab-bar'
+(defun counsel--tab-name (tab)
+  "Return a name for a tab."
+  (cdr (car (cdr tab))))
+
+;;;###autoload
+(defun counsel-tab-bar ()
+  "Switch between tabs."
+  (interactive)
+  (ivy-read
+   "Select tab: "
+   (mapcar #'counsel--tab-name (funcall tab-bar-tabs-function))
+   :Keymap counsel-mode-map
+   :preselect (counsel--tab-name (tab-bar--current-tab))
```

Comment:
Please avoid calling internal functions like `tab-bar--current-tab`, as they can come and go and change behaviour without warning. Instead, can't you use the `current-tab` entry returned by `tab-bar-tabs-function`?

Stepping even further back, why preselect the current tab? Doesn't switching to a tab imply that the user wants to switch to a different tab?